imap_processing/codice/codice_l1a_lo_species.py

```
@@ -0,0 +1,286 @@
+"""CoDICE Lo Species L1A processing functions."""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.codice import constants
+from imap_processing.codice.decompress import decompress
+from imap_processing.codice.utils import (
+    CODICEAPID,
+    ViewTabInfo,
+    calculate_acq_time_per_step,
+    get_codice_epoch_time,
+    get_collapse_pattern_shape,
+    get_view_tab_info,
+    read_sci_lut,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def l1a_lo_species(unpacked_dataset: xr.Dataset, lut_file: Path) -> xr.Dataset:
+    """
+    L1A processing code.
+
+    Parameters
+    ----------
+    unpacked_dataset : xarray.Dataset
+        The decompressed and unpacked data from the packet file.
+    lut_file : pathlib.Path
+        Path to the LUT (Lookup Table) file used for processing.
+
+    Returns
+    -------
+    xarray.Dataset
+        The processed L1A dataset for the given species product.
+    """
+    # Get these values from unpacked data. These are used to
+    # lookup in LUT table.
+    table_id = np.unique(unpacked_dataset["table_id"].values)[0]
+    view_id = np.unique(unpacked_dataset["view_id"].values)[0]
+    apid = np.unique(unpacked_dataset["pkt_apid"].values)[0]
+    plan_id = np.unique(unpacked_dataset["plan_id"].values)[0]
+    plan_step = np.unique(unpacked_dataset["plan_step"].values)[0]
+
+    logger.info(
+        f"Processing species with - APID: {apid}, View ID: {view_id}, "
+        f"Table ID: {table_id}, Plan ID: {plan_id}, Plan Step: {plan_step}"
+    )
+
+    # ========== Get LUT Data ===========
+    # Read information from LUT
+    sci_lut_data = read_sci_lut(lut_file, table_id)
+    if sci_lut_data is None:
+        raise ValueError(f"SCI-LUT file does not have data for table ID {table_id}.")
```

Raise this error from within the read_sci_lut function instead of returning None and requiring handling externally.